src/axom/primal/geometry/Octahedron.hpp

```diff
@@ -217,6 +217,28 @@ class Octahedron
     return (matchedcount == NUM_OCT_VERTS);
   }
 
+  /*!
+   * \brief Determines whether the octahedron has duplicate vertices.
+   * \return True if there are duplicate vertices; False otherwise.
+   */
+  AXOM_HOST_DEVICE
+  bool has_duplicate_vertices() const
+  {
+    for(int i = 0; i < 6; i++)
+    {
+      for(int j = i + 1; j < NUM_OCT_VERTS; j++)
+      {
+        // operator= for Point does not want to play nice...
+        if(m_points[i][0] == m_points[j][0] &&
+           m_points[i][1] == m_points[j][1] && m_points[i][2] == m_points[j][2])
```

Comment:
Outside the scope of this PR, but @bmhan12 (or @agcapps or others) -- should this function be checking for exact equality? Or should it check that the points are within some `eps` tolerance/distance of each other?

Reply:
> should this function be checking for exact equality? Or should it check that the points are within some eps tolerance/distance of each other?

I think a tolerance makes sense here, and makes the function more useable now that it is part of the `Octahedron `class.

Reply:
I agree with @bmhan12 .  I also agree that it is probably outside this PR's scope.